### PR TITLE
publiccloud: Use vaultproject.io to create credentials

### PR DESCRIPTION
**Pull request by cfconrad**
Use REST API of vault server to get temporal credentials. To make it
work you need to specify:
 _SECRET_PUBLIC_CLOUD_REST_URL
 _SECRET_PUBLIC_CLOUD_REST_USER
 _SECRET_PUBLIC_CLOUD_REST_PW

Each provider need a valid vault configuration in the following path:
AWS:   /aws/creds/openqa-role
GCE:   /gcp/key/openqa-role
AZURE: /azure/creds/openqa-role

- Related ticket: https://progress.opensuse.org/issues/42155
- Verification run:
  - Azure: http://cfconrad-vm.qa.suse.de/tests/3340
  - Google: http://cfconrad-vm.qa.suse.de/tests/3345
  - AWS: http://cfconrad-vm.qa.suse.de/tests/3342 

@asmorodskyi @jlausuch WDYT?

### CODE REVIEW

File: lib/publiccloud/azure.pm [hunk 1]
```diff
@@ -36,6 +37,35 @@ sub init {
...
+    record_info('INFO', 'Get credentials from VAULT server.');
+    my $res = $self->vault_api('/v1/azure/creds/openqa-role', method => 'get');
+    $self->vault_lease_id($res->{lease_id});
+    $self->key_id($res->{data}->{client_id});
+    $self->key_secret($res->{data}->{client_secret});
+
+    $res = $self->vault_api('/v1/secret/azure/openqa-role', method => 'get');
+    $self->tenantid($res->{data}->{tenant_id});
+    $self->subscription($res->{data}->{subscription_id});
+    die('Failed to retrieve key') unless (defined($self->key_id) && defined($self->key_secret));
```
Comment (asmorodskyi) [inline lib/publiccloud/azure.pm]: and we are ok if getting tenant_id or subscription_id is missing ?

File: lib/publiccloud/provider.pm [hunk 2]
```diff
@@ -314,25 +316,92 @@ sub terraform_destroy {
...
+=head2 vault_revoke
+
+Revoke a previous retrieved credential
+=cut
+sub vault_revoke {
+    my ($self) = @_;
+
+    return unless (defined($self->vault_lease_id));
+
+    $self->vault_api('/v1/sys/leases/revoke', method => 'post', data => {lease_id => $self->vault_lease_id});
```
Comment (asmorodskyi) [inline lib/publiccloud/provider.pm]: what will happen if this call will fail ? or this would be handled by some autoclean mechanism on server side ?

**Comment by asmorodskyi** [comment]
LGTM

**Comment by jlausuch** [comment]
LGTM

**Pull request merged**